packages/financial-templates-lib/src/price-feed/utils.js

```diff
@@ -36,36 +37,41 @@ exports.BlockHistory = (getBlock, blocks = []) => {
   function getClosestBefore(timestamp) {
     const index = lodash.sortedIndexBy(blocks, { timestamp }, "timestamp");
     // need to check for an exact match in this case, otherwise go to the older block
-    if (blocks[index].timestamp === timestamp) return blocks[index];
+    if (blocks[index] && blocks[index].timestamp === timestamp) return blocks[index];
```

Comment:
When running the SyntheticPegMonitor, I noticed that an error would be thrown here because `index === blocks.length` which makes `blocks[index] === undefined` and `undefined.timestamp` throws an error. Therefore, I added a check to this if statement to make sure that `blocks[index]` exists.